Add possible errors to LSIF docs (#6657)
sourcegraph/sourcegraph: doc/user/code_intelligence/lsif_quickstart.md

If successful, you'll see the following message:

```bash
Repository: <location of repository>
Commit: <commit sha for generated lsif data>
File: <LSIF data file>
Root: <??>

LSIF dump successfully uploaded. It will be converted asynchronously.
To check the status, visit <link to your Sourcegraph instance LSIF status>
```

If an error occurred, you'll see it in the response. Possible errors include:

- Unknown repository (404)
- Invalid commit (404)

> - Invalid auth if LSIF enforce auth is set to true (401 for an invalid token or 404 if repository cannot be found on github)

Reviewer: ```suggestion
- Invalid auth if `lsifEnforceAuth` is `true` (401 for an invalid token or 404 if the repository cannot be found on GitHub.com)
```

- Unexpected errors (500s) 

> - You'll see the job fail in the UI if the LSIF data is bad (malformed indexer) or we encounter a problem processing (e.g. system level bug, flaky connections, etc)

Reviewer: ```suggestion
- LSIF processing failures for a repository are listed in **Repository settings > Code intelligence > Activity for this repository**. Failures can occur if the LSIF data is invalid (e.g., malformed indexer), or problems were encountered during processing (e.g., system-level bug, flaky connections, etc).
```
